doc/source/data/dataset-internals.rst

```diff
@@ -64,19 +64,17 @@ Execution
 
 This section covers Dataset execution modes and performance considerations.
 
-Lazy Execution Mode
-~~~~~~~~~~~~~~~~~~~
-
-By default, most Datasets operations are eager, which provides a simpler iterative
-development experience. Datasets also has a lazy execution mode that can offer
-improved performance due to stage fusion optimizations.
-
-Lazy execution mode can be enabled by calling
-:meth:`ds = ds.lazy() <ray.data.Dataset.lazy()>`, which
-returns a Dataset whose all subsequent operations will be lazy. These operations
-won't be executed until the dataset is consumed or
-:meth:`ds.fully_executed() <ray.data.Dataset.fully_executed>` is called to manually
-trigger execution.
+Lazy Execution
+~~~~~~~~~~~~~~
+
+By default, most Datasets operations are lazy, with execution triggered via "sink"
+APIs, such as consuming (:meth:`ds.iter_batches() <ray.data.Dataset.iter_batches>`),
+writing (:meth:`ds.write_parquet() <ray.data.Dataset.write_parquet>`), certain
+transformations (:meth:`ds.union() <ray.data.Dataset.union>` or
+:meth:`ds.zip() <ray.data.Dataset.zip>`), or manually triggering via
```

Comment:
zip() is lazy: https://sourcegraph.com/github.com/ray-project/ray@6b081e9/-/blob/python/ray/data/dataset.py?L2069